src/virtualdom/Fragment/prototype/rebind.js

```diff
@@ -3,7 +3,9 @@ import { assignNewKeypath } from 'shared/keypaths';
 export default function Fragment$rebind ( oldKeypath, newKeypath ) {
 
 	// assign new context keypath if needed
-	assignNewKeypath( this, 'context', oldKeypath, newKeypath );
+	if ( !this.owner || this.owner.hasContext ) {
+		assignNewKeypath( this, 'context', oldKeypath, newKeypath );
+	}
```

Comment:
I tried that originally, but it failed a test that expected the fragment to grow a context in particular circumstances.